### PR TITLE
Removing file autoload_files.php was added (#4254)

### PR DESCRIPTION
**Pull request by voroks**
in cases of removing package (#4254) or removing autoload flag of package: if list of files to load is empty, file autoload_files.php will be removed

### CODE REVIEW

File: src/Composer/Autoload/AutoloadGenerator.php [hunk 1]
```diff
@@ -255,9 +255,12 @@ public static function autoload(\$class)
         if ($includePathFile = $this->getIncludePathsFile($packageMap, $filesystem, $basePath, $vendorPath, $vendorPathCode52, $appBaseDirCode)) {
             file_put_contents($targetDir.'/include_paths.php', $includePathFile);
         }
```
Comment (Seldaek) [inline src/Composer/Autoload/AutoloadGenerator.php]: How about doing the same for the include_paths.php file above? Might as well fix both at once.

File: src/Composer/Autoload/AutoloadGenerator.php [hunk 2]
```diff
@@ -255,9 +255,12 @@ public static function autoload(\$class)
...
+	$includeFilesFilePath = $targetDir.'/autoload_files.php';
         if ($includeFilesFile = $this->getIncludeFilesFile($autoloads['files'], $filesystem, $basePath, $vendorPath, $vendorPathCode52, $appBaseDirCode)) {
-            file_put_contents($targetDir.'/autoload_files.php', $includeFilesFile);
-        }
+            file_put_contents($includeFilesFilePath, $includeFilesFile);
+        } else if (file_exists($includeFilesFilePath) === true) {
+	    unlink($includeFilesFilePath);
+	}
```
Comment (stof) [inline src/Composer/Autoload/AutoloadGenerator.php]: wrong indentation here

File: src/Composer/Autoload/AutoloadGenerator.php [hunk 3]
```diff
@@ -255,9 +255,12 @@ public static function autoload(\$class)
         if ($includePathFile = $this->getIncludePathsFile($packageMap, $filesystem, $basePath, $vendorPath, $vendorPathCode52, $appBaseDirCode)) {
             file_put_contents($targetDir.'/include_paths.php', $includePathFile);
         }
+	$includeFilesFilePath = $targetDir.'/autoload_files.php';
         if ($includeFilesFile = $this->getIncludeFilesFile($autoloads['files'], $filesystem, $basePath, $vendorPath, $vendorPathCode52, $appBaseDirCode)) {
-            file_put_contents($targetDir.'/autoload_files.php', $includeFilesFile);
-        }
+            file_put_contents($includeFilesFilePath, $includeFilesFile);
+        } else if (file_exists($includeFilesFilePath) === true) {
```
Comment (stof) [inline src/Composer/Autoload/AutoloadGenerator.php]: no need for ``=== true``. Turning a boolean into another boolean is not needed

Comment (voroks) [inline src/Composer/Autoload/AutoloadGenerator.php]: Thanks for your feedback. Code style standard in our company requires explicit comparison and indentation with using tabulation. Sorry for my inattention :)

File: src/Composer/Autoload/AutoloadGenerator.php (hunk 1 above)
Comment (voroks) [inline src/Composer/Autoload/AutoloadGenerator.php]: On the way

**Comment by Seldaek** [comment]
Thanks for the updates, looks good now!

**Pull request merged**